pkg/controller/common/tracing/transaction.go

```diff
@@ -8,16 +8,19 @@ import (
 	"context"
 
 	"go.elastic.co/apm/v2"
-	"k8s.io/apimachinery/pkg/types"
 )
 
+const ReconciliationTxType = "reconciliation"
+const PeriodicTxType = "periodic"
+const RunOnceTxType = "run-once"
```

Comment:
Nit:
```suggestion
const (
	ReconciliationTxType = "reconciliation"
	PeriodicTxType       = "periodic"
	RunOnceTxType        = "run-once"
)
```

Reply:
Have you considered using a custom `TxType` type to avoid that we can invert the `txType` and the `txName` which are both of type `string`?